applications/visualizer/frontend/src/components/Workspace.tsx

```diff
@@ -1,53 +1,65 @@
-import {addWidget} from '@metacell/geppetto-meta-client/common/layout/actions';
-import '@metacell/geppetto-meta-ui/flex-layout/style/dark.scss';
-import theme from '../theme';
-import {useGlobalContext} from "../contexts/GlobalContext.tsx";
-import {rightComponentWidget, threeDViewerWidget} from "../layout-manager/widgets.ts";
-
+import { useDispatch, useSelector } from "react-redux";
+import React, { useEffect, useState } from "react";
+import { ThemeProvider } from "@mui/material/styles";
+import { Box, CircularProgress, CssBaseline } from "@mui/material";
+import { addWidget } from "@metacell/geppetto-meta-client/common/layout/actions";
+import "@metacell/geppetto-meta-ui/flex-layout/style/dark.scss";
+import theme from "../theme";
+import { useGlobalContext } from "../contexts/GlobalContext.tsx";
+import {
+  rightComponentWidget,
+  threeDViewerWidget,
+} from "../layout-manager/widgets.ts";
+import Layout from "./ViewerContainer/Layout.tsx";
 
 function Workspace() {
+  const dispatch = useDispatch();
+  const { workspaces } = useGlobalContext();
 
-    const dispatch = useDispatch();
-    const {workspaces} = useGlobalContext();
-
-    const workspaceId = useSelector(state => state.workspaceId);
-    const [LayoutComponent, setLayoutComponent] = useState<React.ComponentType | undefined>(undefined);
-
-
-    const workspace = workspaces[workspaceId]
+  const workspaceId = useSelector((state) => state.workspaceId);
+  const [LayoutComponent, setLayoutComponent] = useState<
+    React.ComponentType | undefined
+  >(undefined);
+  const [sidebarOpen, setSidebarOpen] = React.useState(true);
 
-    useEffect(() => {
-        if (LayoutComponent === undefined) {
-            if (workspace.layoutManager) {
-                setLayoutComponent(workspace.layoutManager.getComponent());
-            }
-        }
-    }, [LayoutComponent, workspace.layoutManager])
+  const workspace = workspaces[workspaceId];
 
-    useEffect(() => {
-        dispatch(addWidget(threeDViewerWidget()));
-        dispatch(addWidget(rightComponentWidget()));
-    }, [LayoutComponent, dispatch])
+  useEffect(() => {
+    if (LayoutComponent === undefined) {
+      if (workspace.layoutManager) {
+        setLayoutComponent(workspace.layoutManager.getComponent());
+      }
+    }
+  }, [LayoutComponent, workspace.layoutManager]);
 
+  useEffect(() => {
+    dispatch(addWidget(threeDViewerWidget()));
+    dispatch(addWidget(rightComponentWidget()));
+  }, [LayoutComponent, dispatch]);
 
-    const isLoading = LayoutComponent === undefined
-    return (
-        <>
-            <ThemeProvider theme={theme}>
-                <CssBaseline/>
-                {isLoading ?
-                    <CircularProgress/> :
-                    <Box className="layout-manager-container">
-                        <LayoutComponent/>
-                    </Box>
-                }
-            </ThemeProvider>
-        </>
-    )
+  const isLoading = LayoutComponent === undefined;
```

Comment:
Same here, would it be interesting to see if we can use `Suspens`?

Reply:
@aranega for this one we can use the ```Suspens``` as there is a loading state until we get the data, but for the previous one it's more a conditional rendering between 2 components that we have, I only used it here as it's more suitable

Reply:
fyi @aranega the crashing was because the initial state of the LayoutComponent was undefined, there is a possible solution to put the initial state as an empty box

```  const [LayoutComponent, setLayoutComponent] = useState<React.ComponentType>(Box);```

this will fix the crashing issue, but in that case I don't think that we will need the ```Suspense``` tag, as the component will be there in all cases, either empty or with the content, will never take the loading fallback, that's what I think according to the logic, correct me here if i'm wrong, and thanks for the suggestion :)

Reply:
fyi @aranega the crashing was because the initial state of the LayoutComponent was undefined, there is a possible solution to put the initial state as a Loading component 

```    const [LayoutComponent, setLayoutComponent] = useState<React.ComponentType>(() => LoadingComponent);
```

this will fix the crashing issue and will show the loading as an initial state until we have the LayoutComponent,  please let me know if there is a better way, and thanks for the suggestion :)